### DISCUSSION TITLE
`wezterm cli`'s missing command

### DISCUSSION THREAD

**Discussion by cassepipe** [wezterm/wezterm#3350] [Ideas]
I came to `wezterm` hoping for full pane/tab management from the command line but there seem to be one missing command :
```bash
wezterm cli activate-tab-direction
```
Can we have it (please) ? Or is there a technical reason behind its non-implementation (Let's hope not) ?

I'd implement it myself but the only language I am confident coding with for now is C (unexpected I know).

A nice addition would be for the `wezterm cli activate-pane-direction` to actually (optionnally?) change tabs where theres's no panes in the current tab.

What do you think ?

**Comment by cassepipe** [comment]
Yes, I am using aliases, I am not crazy.

And since you didn't ask, navigation commands are easily removed from `zsh`'s history by setting an option and prefixing your commands/aliases with a space or two (or more) 

```zsh
setopt histignorespace
```
Fun fact : uppercase or lowercase or underscores do not matter in options's names so you might as well make it look authoritarian :
```zsh
setoption HIST_IGNORE_SPACE
```
Now  in your `.zshrc` you can 
```zsh
alias w="   wezterm cli"
alias wp="  wezterm cli activate-new-pane"
#etc.
```
There is `bash` 's `HISTCONTROL=ignorespace` environment variable that does the same thing and pattern based options too.

**Comment by wez** [comment]
Do you mean that you'd like a `wezterm cli` equivalent to https://wezfurlong.org/wezterm/config/lua/keyassignment/ActivateTabRelative.html ?
Please file an enhancement issue for that.

**Comment by cassepipe** [comment]
Here is the enhancement issue: https://github.com/wez/wezterm/issues/3352
